src/main/webapp/go.js

```diff
@@ -200,10 +206,135 @@ function toggleHintButton(hide) { //eslint-disable-line
 function getHint() { //eslint-disable-line
   const nextHint = hunt.getNextHint();
   if (nextHint != '') {
-    updateMessage(HINT_DISPLAY, nextHint);
+    updateMessage(HINT_DISPLAY, 'Hint #' + (hunt.getHintIndex()) +
+        ': ' + nextHint);
+  } else {
+    if (hunt.getPlaceID() == -1) {
+      autoGenerateHints();
```

Comment:
Why is `autoGenerateHints()` only called if the user hasn't started the hunt yet? Is it possible that this method won't be called before the user beings, and if so, will this cause problems for `getNextAutoHint()`, e.g. the auto hints might not exist?

Reply:
Update: Maybe placeID is the Google Places API ID? If this code is correct, could you add some comments to explain what it's doing?

Reply:
Update 2: It might be cleaner to move this branching logic and into `getNextAutoHint()`. Here's a pseudocode version of what I'm thinking:

```
function getNextAutoHint() {
  if (hunt.getPlaceID() == -1) {
    autoGenerateHints();    
  } else {
    const random = Math.random();
    // And so on...
  }
}
```

Reply:
Yes, getPlaceID() refers to the unique ID for a destination provided by the Places API -- I added comments to clarify this, and restructured the branching logic.